fbchat/models.py

```diff
@@ -4,471 +4,169 @@
 import enum
 
 
-class FBchatException(Exception):
-    """Custom exception thrown by fbchat. All exceptions in the fbchat module inherits this"""
-
-class FBchatFacebookError(FBchatException):
-    #: The error code that Facebook returned
-    fb_error_code = None
-    #: The error message that Facebook returned (In the user's own language)
-    fb_error_message = None
-    #: The status code that was sent in the http response (eg. 404) (Usually only set if not successful, aka. not 200)
-    request_status_code = None
-    def __init__(self, message, fb_error_code=None, fb_error_message=None, request_status_code=None):
-        super(FBchatFacebookError, self).__init__(message)
-        """Thrown by fbchat when Facebook returns an error"""
-        self.fb_error_code = str(fb_error_code)
-        self.fb_error_message = fb_error_message
-        self.request_status_code = request_status_code
-
-class FBchatUserError(FBchatException):
-    """Thrown by fbchat when wrong values are entered"""
+class Enum(enum.Enum):
+    """Used internally by fbchat to support enumerations"""
+    def __repr__(self):
+        # For documentation:
+        return '{}.{}'.format(type(self).__name__, self.name)
 
-class Thread(object):
-    #: The unique identifier of the thread. Can be used a `thread_id`. See :ref:`intro_threads` for more info
-    uid = None
-    #: Specifies the type of thread. Can be used a `thread_type`. See :ref:`intro_threads` for more info
-    type = None
-    #: A url to the thread's picture
-    photo = None
-    #: The name of the thread
-    name = None
-    #: Timestamp of last message
-    last_message_timestamp = None
-    #: Number of messages in the thread
-    message_count = None
-    def __init__(self, _type, uid, photo=None, name=None, last_message_timestamp=None, message_count=None):
-        """Represents a Facebook thread"""
-        self.uid = str(uid)
-        self.type = _type
-        self.photo = photo
-        self.name = name
-        self.last_message_timestamp = last_message_timestamp
-        self.message_count = message_count
 
-    def __repr__(self):
-        return self.__unicode__()
+class FacebookError(Exception):
+    """Thrown by fbchat when Facebook returns an error
+
+    Attributes:
+        fb_error_code (int): The error code that Facebook returned
+        fb_error_message: A localized error message that Facebook returned
+    """
 
-    def __unicode__(self):
-        return '<{} {} ({})>'.format(self.type.name, self.name, self.uid)
+
+class Thread(object):
+    """Represents a Facebook chat-thread
+
+    Attributes:
+        id (int): The unique identifier of the thread
```

Comment:
`Thread.id` or `Thread.uid`? `Thread.id` doesn't actually conflict with the builtin `id`, and is more clean IMO

Reply:
Also, I'm advocating for integers, though maybe we should leave it as strings?